doc/api/assert.md

```diff
@@ -2539,6 +2503,111 @@ assert.throws(throwingFirst, /Second$/);
 // AssertionError [ERR_ASSERTION]
 ```
 
+## `assert.matchObject(actual, expected[, message])`
+
+<!-- YAML
+added: v0.1.21
+changes:
+  - version:
+      - v16.0.0
+      - v14.18.0
+    pr-url: https://github.com/nodejs/node/pull/38113
+    description: Equivalence assertion test between two objects.
+  - version: v14.0.0
+    pr-url: https://github.com/nodejs/node/pull/30766
+    description: NaN is now treated as being identical if both sides are
+                 NaN.
+-->
+
+- `actual` {any}
+- `expected` {any}
+- `message` {string|Error}
+
+**Strict assertion mode**
+
+An alias of [`assert.matchObjectStrict()`][].
+
+**Legacy assertion mode**
+
+> Stability: 3 - Legacy: Use [`assert.matchObjectStrict()`][] instead.
+
+Tests shallow, coercive inequality with the [`!=` operator][]. `NaN` is
+specially handled and treated as being identical if both sides are `NaN`.
+
+```mjs
+import assert from 'node:assert';
+
+assert.matchObject({ a: 1 }, { a: 1 });
+// OK
+
+assert.matchObject({ b: 1 }, { b: 2 });
+// AssertionError: 1 != 2
+
+assert.matchObject({ c: 1 }, { c: '1' });
+// OK
+```
+
+```cjs
+const assert = require('node:assert');
+
+assert.matchObject({ a: 1 }, { a: 1 });
+// OK
+
+assert.matchObject({ b: 1 }, { b: 2 });
+// AssertionError: 1 != 2
+
+assert.matchObject({ c: 1 }, { c: '1' });
+// OK
+```
+
+If the values or keys are equal, an [`AssertionError`][] is thrown with a `message`
+property set equal to the value of the `message` parameter. If the `message`
+parameter is undefined, a default error message is assigned. If the `message`
+parameter is an instance of an [`Error`][] then it will be thrown instead of the
+`AssertionError`.
+
+## `assert.matchObjectStrict(actual, expected[, message])`
+
+<!-- YAML
+added: v0.1.21
+changes:
+  - version: v10.0.0
+    pr-url: https://github.com/nodejs/node/pull/17003
+    description: Strict comparisson of two objects.
+-->
+
+- `actual` {any}
+- `expected` {any}
+- `message` {string|Error}
+
+Tests strict equality between the `actual` and `expected` parameters.
```

Comment:
That's hardly what's happening, is it? At least I find the wording very confusing (I would expect the same behavior as `strictEqual` with the current wording), we should explain in more details what's happening here, what's being compared: is it only own properties? is the prototype important?